pkg/operator/api/userconfig/apis.go

```diff
@@ -119,17 +119,23 @@ func (api *API) Validate() error {
 	if yaml.StartsWithEscapedAtSymbol(api.Model) {
 		api.ModelFormat = TensorFlowModelFormat
 	} else {
-		if api.ModelFormat == UnknownModelFormat {
-			return errors.Wrap(cr.ErrorMustBeDefined(), Identify(api), ModelFormatKey)
-		}
-
 		if !aws.IsValidS3Path(api.Model) {
 			return errors.Wrap(ErrorInvalidS3PathOrResourceReference(api.Model), Identify(api), ModelKey)
 		}
 
 		if ok, err := aws.IsS3PathFileExternal(api.Model); err != nil || !ok {
 			return errors.Wrap(ErrorExternalNotFound(api.Model), Identify(api), ModelKey)
 		}
+
+		if api.ModelFormat == UnknownModelFormat {
+			if strings.HasSuffix(api.Model, ".onnx") {
+				api.ModelFormat = ONNXModelFormat
+			} else if strings.HasSuffix(api.Model, ".zip") {
+				api.ModelFormat = TensorFlowModelFormat
+			} else {
+				return errors.Wrap(cr.ErrorMustBeDefined(), Identify(api), ModelFormatKey)
```

Comment:
Do you think that we should mention in the error message that changing the suffix to `.zip` or `.onnx` would also work?

Reply:
Were you thinking a message like this?
`path to model should end in .zip suffix for tensorflow models, .onnx for ONNX models or manually specify model_format key`

Reply:
Yeah sounds good (maybe add comma after "ONNX models")